core-bundle/src/Security/Voter/DataContainer/TableAccessVoter.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Security\Voter\DataContainer;
+
+use Contao\CoreBundle\Security\ContaoCorePermissions;
+use Contao\CoreBundle\Security\DataContainer\DataContainerSubject;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Authorization\Voter\CacheableVoterInterface;
+use Symfony\Component\Security\Core\Security;
+
+/**
+ * @internal
+ */
+class TableAccessVoter implements CacheableVoterInterface
+{
+    public function __construct(private readonly Security $security)
+    {
+    }
+
+    public function supportsAttribute(string $attribute): bool
+    {
+        return \in_array($attribute, [
+            ContaoCorePermissions::DC_ACTION_CREATE,
+            ContaoCorePermissions::DC_ACTION_EDIT,
+            ContaoCorePermissions::DC_ACTION_COPY,
+        ], true);
+    }
+
+    public function supportsType(string $subjectType): bool
+    {
+        return DataContainerSubject::class === $subjectType;
+    }
+
+    /**
+     * @param DataContainerSubject $subject
+     */
+    public function vote(TokenInterface $token, $subject, array $attributes): int
+    {
+        foreach ($attributes as $attribute) {
+            if (!$this->supportsAttribute($attribute)) {
+                continue;
+            }
+
+            $hasNotExcluded = false;
+
+            foreach (($GLOBALS['TL_DCA'][$subject->table]['fields'] ?? []) as $config) {
```

Comment:
Needs to call `loadDataContainer()` here :)

Reply:
it intentionally does not (maybe I should add a comment). I would expect someone has loaded the DC when checking permission for the DC. However, if the DC is not loaded, no harm will happen, the permission voter will simply deny if none of the table fields was given permission. The only case this could go wrong is if no DC is loaded and all fields would be set to `exclude = false`.

Reply:
Yes comment!

Reply:
added in ec867ca30275216fdee18d7f68f546383aaaf5ff